include/pluginlib/class_loader_imp.hpp

```diff
@@ -507,8 +506,22 @@ std::string ClassLoader<T>::getName(const std::string & lookup_name)
 /***************************************************************************/
 {
   // remove the package name to get the raw plugin name
-  std::vector<std::string> result = pluginlib::impl::split(lookup_name, "/|:");
-  return result.back();
+  auto last_slash = lookup_name.find_last_of('/');
+  auto last_colon = lookup_name.find_last_of(':');
+  if (last_slash == std::string::npos && last_colon == std::string::npos) {
+    // no matches
+    return lookup_name;
+  }
+  if (last_slash == std::string::npos) {
+    // only colon matched
+    return lookup_name.substr(last_colon + 1);
+  }
+  if (last_colon == std::string::npos) {
+    // only slash matched
+    return lookup_name.substr(last_slash + 1);
+  }
+  // both matched, return shorter suffix
+  return lookup_name.substr(std::max(last_slash, last_colon) + 1);
```

Comment:
Thank you for the PR!

Please add unit tests for `ClassLoader<T>::getName` so we can make sure the code works the same before and after the refactor 
 https://github.com/ros/pluginlib/blob/rolling/test/utest.cpp

Reply:
Please also add `#include <algorithm>` for max to make the linter happy: https://build.ros2.org/job/Rpr__pluginlib__ubuntu_noble_amd64/17/testReport/junit/pluginlib/cpplint/build_include_what_you_use__4____tmp_ws_src_pluginlib_include_pluginlib_class_loader_imp_hpp_524_/